llvm/lib/Target/RISCV/RISCVCodeGenPrepare.cpp

```diff
@@ -155,6 +163,43 @@ bool RISCVCodeGenPrepare::visitIntrinsicInst(IntrinsicInst &I) {
   return true;
 }
 
+bool RISCVCodeGenPrepare::expandVPStrideLoad(IntrinsicInst &II) {
+  if (ST->hasOptimizedZeroStrideLoad())
+    return false;
+
+  Value *BasePtr, *VL;
+  using namespace PatternMatch;
+  if (!match(&II, m_Intrinsic<Intrinsic::experimental_vp_strided_load>(
+                      m_Value(BasePtr), m_Zero(), m_AllOnes(), m_Value(VL))))
+    return false;
+
+  if (!isKnownNonZero(VL, {*DL, DT, nullptr, &II}))
+    return false;
+
+  auto *VTy = cast<VectorType>(II.getType());
+
+  // FIXME: Support fixed vector types.
+  if (!isa<ScalableVectorType>(VTy))
+    return false;
```

Comment:
What would you think about ignoring the EVL and just doing `Builder.CreateVectorSplat` so we can handle both fixed-length and scalable vectors?

My reason being that I think a lot of these splats will end up getting matched as a scalar argument to .vx instructions anyway. And I think it's more important that we end up matching .vx instructions for fixed-length vectors even if we might have some extra vl toggles due to the EVL mismatch

Reply:
The original motivation for this patch was for our downstream vectorizer which generates in-loop vsetvlis and vp.strided.load directly. I don't want more VL toggles in the loop.

We could use Luke's original DAG combine patch for the fixed vector case since the EVL there is likely to be a constant.

We could add vp.splat. This isn't the first time I thought about adding it.

We could make riscv.vmv.v.x work with fixed vectors.

Reply:
I think vp.splat makes sense to have, if we already have vp.splice and vp.reverse for the other shufflevector patterns.

I think we should go ahead with this PR with or without the fixed vector case though, I would prefer it over the DAG combine since its more powerful